models/demos/tg/llama3_70b/tt/llama_attention_galaxy.py

```diff
@@ -467,3 +491,194 @@ def attn_selfout(
         )
 
         return attn_output
+
+    def prefill_forward(
+        self,
+        xs,
+        rot_mats,
+        attn_masks,
+        user_id: int = 0,
+    ):
+        query_layer, key_layer, value_layer = self.prefill_attn_qkv(xs, rot_mats)
+        attn_outputs = self.prefill_attn_mqa(query_layer, key_layer, value_layer, attn_masks, user_id)
+        return self.prefill_attn_selfout(attn_outputs)
+
+    def prefill_attn_qkv(
+        self,
+        xs,
+        rot_mats,
+    ):
+        assert xs.shape[1] == 1, "batch must be 1"
+        assert xs.shape[2] % 128 == 0 and xs.shape[2] > 0, "Seqlen must be divisible by 128"
+        _, _, seq_len, _ = xs.shape
+
+        max_mm_seq_len = self.model_config["MAX_MM_SEQ_LEN"]
+        batch_dim = 1 if seq_len < max_mm_seq_len else seq_len // max_mm_seq_len  # Find the division factor
+
+        # TODO: add batch dim for looping along it in case of seq_len > max_mm_seq_len
```

Comment:
Is this still a todo? Looks like the batch dim is already based on max_mm_seq_len and used to reshape the activation?

Reply:
This is to have in mind when working on efficient implementation for larger seq_lens.